genai-perf/tests/test_tokenizer.py

```diff
@@ -38,6 +42,16 @@ def test_non_default_tokenizer(self):
         tokenizer_model = "gpt2"
         get_tokenizer(tokenizer_model)
 
+    def test_default_tokenizer_all_args(self):
+        tokenizer_model = DEFAULT_TOKENIZER
+        get_tokenizer(tokenizer_model, False, DEFAULT_TOKENIZER_REVISION)
+
+    def test_non_default_tokenizer_all_args(self):
+        tokenizer_model = "gpt2"
+        get_tokenizer(
+            tokenizer_model, False, "11c5a3d5811f50298f278a704980280950aedb10"
```

Comment:
is it possible to check whether the tokenizer attributes are correctly set? 

e.g. `tokenizer.trust_remote_code == True`

Reply:
Unfortunately, there doesn't seem to be a getter to run this check. The main way to test this would be to have a tokenizer locally or on Hugging Face that requires this to be set to true but then it wouldn't be a unit test (and would expose the test to some security risks). We could alternatively mock the initialization call to ensure it's called with the right args, but then we're just redoing what the function does.

Reply:
sounds good. yeah I was hoping if that was possible but I guess there's no simple way to test this... 

if there's no way to check the changes, is there any point in having these new tests?

Reply:
We'd be testing that the external code doesn't change and that this function doesn't change. These tests would break if:
- The function signature changes
- The HuggingFace tokenizer parameters change
So this would catch a regression but isn't strictly necessary. These are similar to the tests right above them. If you'd like though, I can remove them.

Reply:
I see. Yeah I think passing non default args could reveal if anything changes on the huggingface side. Instead of removing, can we just pass `True` so that we are passing non default for all arguments?
```python
get_tokenizer(
    tokenizer_model, True, "11c5a3d5811f50298f278a704980280950aedb10"
```

Reply:
If we pass True, we expose our GitLab runners to a security risk. The Tokenizer model should be safe with the hash there, but I didn't want to add that risk.

Reply:
makes sense. we can keep it as is